platform/admin/core/admin-core-appservice/src/main/java/org/codice/ddf/admin/application/service/migratable/Operation.java

```diff
@@ -17,13 +17,20 @@
  * A list of operations that can be performed on applications, features, or bundles.
  *
  * <p><i>Note:</i> The order defined here is important and represents the order tasks will be
- * executed.
+ * executed. When it comes to bundles, it is important to have the bundles started in the right
+ * order as such we have to give priority to installation. We will then uninstall and stop
+ * everything else and finish off with starting. This will make sure that we will not have 2 bundles
+ * that might provide the same implementation for an API started at the same time.
+ *
+ * <p>The update operation is a special operation used by features where, for example, feature
+ * requirements needs to be added or removed to mark the feature as required or not.
```

Comment:
❗️ You explained what the update operation does, but could you explain why it's necessary? Even after reading the code and your discussion on the karaf dev group, I don't understand why it's necessary.

Reply:
I added a little bit more doc, I hope that is enough (note to self, Brandan asked for more doc 😄  )